lib/grp.gi

```diff
@@ -30,14 +30,15 @@ InstallImmediateMethod( IsFinitelyGeneratedGroup,
 ##
 #M  IsCyclic( <G> ) . . . . . . . . . . . . . . . . test if a group is cyclic
 ##
-InstallImmediateMethod( IsCyclic, IsGroup and HasGeneratorsOfGroup, 0,
-    function( G )
-    if Length( GeneratorsOfGroup( G ) ) = 1 then
-      return true;
-    else
-      TryNextMethod();
-    fi;
-    end );
+# this is now set when creating the groups
+#InstallImmediateMethod( IsCyclic, IsGroup and HasGeneratorsOfGroup, 0,
+#    function( G )
+#    if Length( GeneratorsOfGroup( G ) ) = 1 then
+#      return true;
+#    else
+#      TryNextMethod();
+#    fi;
+#    end );
```

Comment:
Fine by me.

Ooops, the next method after this one is incorrect: it does
```
if HasGeneratorsOfGroup( G ) and Length( GeneratorsOfGroup(G) ) = 1 then
  SetMinimalGeneratingSet(G,GeneratorsOfGroup(G));
```
which is incorrect if `GeneratorsOfGroup(G)` consists only of an identity element. This is of course currently hidden by the immediate method. Mental note to self: try to find a tests case triggering this (after the immediate is gone), and fix it. (But anybody else feel free to fix this first, I won't work on it today for sure ;-)